config.go

```diff
@@ -139,6 +158,17 @@ func ParseConfig(path string) (*Config, error) {
 	// Store a reference to the path where this config was read from
 	config.Path = path
 
+	// Parse the MaxStale component
+	if raw := config.MaxStaleRaw; raw != "" {
+		stale, err := time.ParseDuration(raw)
+
+		if err == nil {
+			config.MaxStale = stale
+		} else {
+			errs = multierror.Append(errs, fmt.Errorf("max_stale invalid: %v", err))
```

Comment:
In Consul, if the value is less than 1000 we multiply by time.Second. Otherwise `max_stale = 5` means 5 nanoseconds, which is pretty useless and causes every query to be done twice. I'd add a similar guard.